pyvista/pyvista
# Release 0.28

### Release 0.28

The last release of `pyvista` was 0.27.0 back in 8 November.  While work was somewhat light over the holidays, it's picked up recently and I think it's time for a minor release given the recent bug-fixes and features.  There's still room for additional features, but I'd like it out soon, ideally this Sunday, but I'm fine with pushing it back further.

As for the current pending PRs, I've moved some milestone items to 0.29, which I've set tentatively to ~28 Feb~ 12 Mar.

For all who contributed, thank you!  I'm always amazed at the number of stars we've accumulated and the number of people who contribute.  While I'm not sure about the accuracy of the statistics, [pypistats](https://pypistats.org/packages/pyvista) reports around 7k downloads for last month.

---


### Release Candidate Testing
@pyvista/developers and any willing testers, the latest release candidate package can be downloaded from pypi with:

```
pip install pyvista==0.28.0rc3
```

See:

https://pypi.org/project/pyvista/0.28.0rc3/

I really like how ``numpy`` is putting out pre-release wheels, and it's a something we should implement as well.


### Proposed release notes:

Please let me know if anything here needs to be updated.

---

# Release 0.28: Minor API Changes and Various Bug-fixes

This release, ``pyvista==0.28.0``, is a minor release that captures a variety of bug-fixes and API additions.

#### Contributors (alphabetical)

- @adeak
- @akaszynski
- @banesullivan
- @Boorhin
- @dcbr
- @GuillaumeFavelier
- @JevinJ
- @Keou0007
- @larsoner
- @mglesser
- @nickvazz
- @rodrigomologni
- @sampotter
- @shakasaki
- @Sinaxist
- @Spectre5
- @tkoyama010
- @whophil
- @yngvem


#### New Features

- Add table of glyphs option to DataSetFilters.glyph (#776)
- Allow picking in notebook with ipyvtk (#996)
- Rename trans_from_matrix to array_from_vtkmatrix, implement inverse (#1110)
- Add slicing, concatenation to StructuredGrid (#1106)
- 🚩 Add option level and quads to function Box (#1103)
- ✨ Add Pyramid function (#1102)
- Add make_tri_mesh helper (#1101)
- Add PolyData intersection filter (#1091)
- Add read_plot3d, other minor reader improvements (#1081)
- Add parallel projection and parallel scale properties to Renderer and BasePlotter (#1021)

#### Documentation

- use pypandoc for better jupyter notebooks from the gallery (#1017)
- Doc/move around (#1000)
- improve camera docs and implementation (#1111)
- :pencil2: Fix typo of docstring (#1104)
- add tuple as type hint for vectors (#1094)
- fix copy button (#1063)
- feat/pyinstaller docs (#1037)

#### API Changes

- Use render window size (#995)
- Add variable length density to voxelize (#1100)
- 📷 Add camera object (#945)
- Add style parameter in text slider widgets (#1093)

#### Bug Fixes

- fix/threshold args (#1014)
- Fix small notation shaking (#1008)
- Fix subplotting with ipyvtk_simple (#994)
- Update hypothesis requirement 5.8.0->6.0.3 (#1115)
- Replace "is 0" comparison with "== 0" in filters.py (#1114)
- Release graphics resources (#1113)
- Ensure that views of pyvista_ndarray remain associated (#1108)
- Fixed disk geometry to rotate and translate properly (#1061)
- Fix double blocking of `plotter.show` in script mode on Windows (#1105)
- Fix "Hide points in add_point_labels is not working" (#1092)
- Fix window closing immediately (#1048)
- Fix/extract points (#1043)
- Fix UnstructuredGrid init from arrays for VTK 8.90 (#1039)
- Fix Cell Types Helper (#1028)
- Add missing no_itk skip to itk polydata point plotting test (#1032)
- Fix `add_arrows` (#1011)
- Bugfix for #1053: `Plotter.update` fails with `interactive_update=True` if `auto_close=False` is not specified (#1056)
- patched edge case for empty mesh (#1030)
- force rewrite faces when flipping normals (#1034)
- fix/default rcParams (#1015)

#### Miscellaneous/Testing

- ENH: Reduce size (#1019)
- Remove Alpha Channel on Image Regression (#1006)
- Clean up example downloads (#1005)
- removing py3.9 check (#1003)
- Image Regression (#998)
- Disable conda 8.1 testing and update agents (#1109)
- write frame needs update (#1099)
- do not import pandas on __init__ (#1029)
